contracts/utils/SignatureValidator.sol

```diff
@@ -1,195 +1,91 @@
         // signature array with invalid type or length. We may as well make
         // it an explicit option. This aids testing and analysis. It is
         // also the initialization value for the enum type.
-        if (signatureType == SignatureType.Illegal) {
-            revert("SignatureValidator#isValidSignature: illegal signature");
-
-            // Signature using EIP712
-        } else if (signatureType == SignatureType.EIP712) {
-            require(_sig.length == 65 || _sig.length == 97, "SignatureValidator#isValidSignature: length 65 or 97 required");
-            r = _sig.readBytes32(0);
-            s = _sig.readBytes32(32);
-            v = uint8(_sig[64]);
-            recovered = ecrecover(_hash, v, r, s);
-            isValid = _signerAddress == recovered;
-            return isValid;
-
-            // Signed using web3.eth_sign() or Ethers wallet.signMessage()
-        } else if (signatureType == SignatureType.EthSign) {
-            require(_sig.length == 65 || _sig.length == 97, "SignatureValidator#isValidSignature: length 65 or 97 required");
-            r = _sig.readBytes32(0);
-            s = _sig.readBytes32(32);
-            v = uint8(_sig[64]);
-            recovered = ecrecover(keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", _hash)), v, r, s);
-            isValid = _signerAddress == recovered;
-            return isValid;
-
-            // Signature verified by wallet contract with data validation.
-        } else if (signatureType == SignatureType.WalletBytes) {
-            isValid = ERC1271_MAGICVALUE == IERC1271Wallet(_signerAddress).isValidSignature(_data, _sig);
-            return isValid;
-
-            // Signature verified by wallet contract without data validation.
-        } else if (signatureType == SignatureType.WalletBytes32) {
-            isValid = ERC1271_MAGICVALUE_BYTES32 == IERC1271Wallet(_signerAddress).isValidSignature(_hash, _sig);
-            return isValid;
-        } else if (signatureType == SignatureType.Wallet) {
-            isValid = isValidWalletSignature(_hash, _signerAddress, _sig);
-            return isValid;
-        }
 
-        // Anything else is illegal (We do not return false because
-        // the signature may actually be valid, just not in a format
-        // that we currently support. In this case returning false
-        // may lead the caller to incorrectly believe that the
-        // signature was invalid.)
-        revert("SignatureValidator#isValidSignature: unsupported signature");
+        revert("SignatureValidator#isValidSignature: illegal signature");
+    } else if (signatureType == SignatureType.EIP712) {
+        // In ealier version, an extra bytes32 is packed with (v,r,s) but the field was never used.
+        // So the _sig was in 97 bytes before.
+        // In order to support standard 65 bytes ECDSA signature and legacy format, still spliting _sig here
+        // instead of passing _sig directly to OZ ECDSA lib.
```

Comment:
I would suggest:
```suggestion
        // To be backward compatible with previous signature format which has (v,r,s) arranged in the order (r,s,v) and with an extra 32 bytes padded in the end, here we just extract the (v,r,s) from the signature and ignore the rest.
```

Reply:
By looking at the code it seems that OZ lib expects the order `v,r,s`: `ECDSA.recover(_hash, v, r, s);`

while we expect the order in the signature to be `r,s,v`?
```solidity
bytes32 r = LibBytes.readBytes32(_sig, 0);
bytes32 s = LibBytes.readBytes32(_sig, 32);
uint8 v = uint8(_sig[64]);
```

Reply:
Oh I mean this function `function recover(bytes32 hash, bytes memory signature)`. The layout of `bytes signature` is (r,s,v).

Reply:
I see what you mean. So we are just ditching the rest of the bytes beyond 65th byte.
```suggestion
        // To be backward compatible with previous signature format which has an extra 32 bytes padded in the end, here we just extract the (r,s,v) from the signature and ignore the rest.
```

Reply:
Updated in 8bcb604